index.js

```diff
@@ -39,9 +43,17 @@ const callLimit = (queue, limit = defLimit) => new Promise((res, rej) => {
     results[c] = queue[c]().then(result => {
       active --
       results[c] = result
-      run()
+      if (!rejected)
+        run()
       return result
-    }, reject)
+    }, (er) => {
+      active --
+      reject(er)
+    }).finally(result => {
```

Comment:
```suggestion
    }).then(result => {
```

Realized this was still here from debugging. Should be a normal `then`